test/functional/test_framework/test_node.py

```diff
@@ -118,22 +118,20 @@ def get_deterministic_priv_key(self):
     def get_mem_rss(self):
         """Get the memory usage (RSS) per `ps`.
 
-        If process is stopped or `ps` is unavailable, return None.
+        Returns None if `ps` is unavailable.
         """
-        if not (self.running and self.process):
-            self.log.warning("Couldn't get memory usage; process isn't running.")
-            return None
+        assert self.running
 
         try:
             return int(subprocess.check_output(
-                "ps h -o rss {}".format(self.process.pid),
-                shell=True, stderr=subprocess.DEVNULL).strip())
+                ["ps", "h", "-o", "rss", "{}".format(self.process.pid)],
+                stderr=subprocess.DEVNULL).strip())
 
-        # Catching `Exception` broadly to avoid failing on platforms where ps
-        # isn't installed or doesn't work as expected, e.g. OpenBSD.
+        # Catching to avoid failing on platforms where ps
+        # isn't installed (SubprocessError) or doesn't work as expected (ValueError), e.g. BSD.
         #
         # We could later use something like `psutils` to work across platforms.
-        except Exception:
+        except (subprocess.SubprocessError, ValueError):
```

Comment:
We could also try detecting/parsing the *BSD-style output, which looks like `"RSS\n 123"`.